packages/react/src/components/Table/TableHeader.tsx

```diff
@@ -11,45 +12,69 @@ interface TableHeaderProps extends ThHTMLAttributes<HTMLTableCellElement> {
   sortDirection?: SortDirection;
   onSort?: () => void;
   className?: string;
+  sortAscendingAnnouncement?: string;
+  sortDescendingAnnoucnement?: string;
 }
 
 const TableHeader = ({
   children,
   sortDirection,
   onSort,
   className,
+  sortAscendingAnnouncement = 'sorted ascending',
+  sortDescendingAnnoucnement = 'sorted descending',
   ...other
-}: TableHeaderProps) => (
-  <th
-    aria-sort={sortDirection}
-    className={classNames('TableHeader', className, {
-      'TableHeader--sort-ascending': sortDirection === 'ascending',
-      'TableHeader--sort-descending': sortDirection === 'descending'
-    })}
-    {...other}
-  >
-    {!!onSort && !!sortDirection ? (
-      <button
-        onClick={onSort}
-        className="TableHeader__sort-button"
-        type="button"
-      >
-        {children}
-        <span aria-hidden="true">
-          {sortDirection === 'none' ? (
-            <Icon type="sort-triangle" />
-          ) : sortDirection === 'ascending' ? (
-            <Icon type="triangle-up" />
-          ) : (
-            <Icon type="triangle-down" />
-          )}
-        </span>
-      </button>
-    ) : (
-      children
-    )}
-  </th>
-);
+}: TableHeaderProps) => {
+  // When the sort direction changes, we want to announce the change in a live region
+  // because changes to the sort value is not widely supported yet
+  // see: https://a11ysupport.io/tech/aria/aria-sort_attribute
+  const [announcement, setAnnouncement] = useState('');
+  useEffect(() => {
+    if (sortDirection !== 'none') {
+      setAnnouncement(
+        sortDirection === 'ascending'
+          ? sortAscendingAnnouncement
+          : sortDescendingAnnoucnement
+      );
+    }
+  }, [sortDirection]);
+  return (
+    <th
+      aria-sort={sortDirection}
+      className={classNames('TableHeader', className, {
+        'TableHeader--sort-ascending': sortDirection === 'ascending',
+        'TableHeader--sort-descending': sortDirection === 'descending'
+      })}
+      {...other}
+    >
+      {!!onSort && !!sortDirection ? (
+        <button
+          onClick={onSort}
+          className="TableHeader__sort-button"
+          type="button"
+        >
+          {children}
+          <span aria-hidden="true">
+            {sortDirection === 'none' ? (
+              <Icon type="sort-triangle" />
+            ) : sortDirection === 'ascending' ? (
+              <Icon type="triangle-up" />
+            ) : (
+              <Icon type="triangle-down" />
+            )}
+          </span>
```

Comment:
For my own edification, would another way to solve this be to make this a live region with labels? Or would that wind up being too verbose when reading through it normally (do screen readers already say what the sort direction is?)

Reply:
It might be too verbose.

When I was testing in JAWS, the behavior was something like:

> Table Column 4 Sorted Ascending, Email, Button

Unfortunately there's a bit of redundancy here, but we probably don't want want the current sort state to be part of the calculated accessible name of the button.

Reply:
Actually, I guess it shouldn't be any more verbose than the other live region. So if we changed this to something like:
```
<span aria-role="status">
  {sortDirection === 'none' ? (
    <Icon type="sort-triangle" />
  ) : sortDirection === 'ascending' ? (
    <Icon type="triangle-up"  aria-label={{sortAscendingAnnouncement}} />
  ) : (
    <Icon type="triangle-down" aria-label={{sortDescendingAnnouncement}} />
  )}
</span>
```

Could we do away with the `announcement` variable?

Reply:
I replied before seeing your reply.

Reply:
There's also the issue with the above being `aria-hidden` so the labels would not be visible to AT users anyway

Reply:
In my example, I removed the `aria-hidden`. I don't think it would be any more verbose than what is currently here and might simplify the code some, but it's not that big a deal.